src/Domain/Direction.hpp

```diff
@@ -43,18 +43,21 @@ class Direction {
   double sign() const noexcept { return (Side::Lower == side_ ? -1.0 : 1.0); }
 
   /// The opposite Direction.
-  Direction<VolumeDim> opposite() const;
+  Direction<VolumeDim> opposite() const noexcept;
+
+  // An array of Directions.
```

Comment:
I'm not sure this comment is useful. Maybe `An array of all logical Directions for a given dimensionality`?